### PR TITLE
⭐️ full IMDSv2 support for Amazon instances

### PR DESCRIPTION
**Pull request by vjeffrey**
fixes https://github.com/mondoohq/cnquery/issues/1813

This change updates the platform detection to work well with IMDSv2 for both Linux and Windows. In addition it slickly tweaks the default platform ids:

- machine-id is disable by default since it is not reliable
- cloud-detect is enabled as default now

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -177,6 +177,15 @@
...
         "status",
       ],
+    },
+    {
+      "name": "cnquery-shell",
+      "type": "go",
+      "request": "launch",
+      "program": "${workspaceRoot}/apps/cnquery/cnquery.go",
+      "args": [
+        "shell", "ssh", "administrator@18.215.249.49", "--password", "8!dVWy&WRh)kgaTHB=7A9Chfw9v9I.94",
```
Comment (preslavgerchev) [inline .vscode/launch.json]: we should remove that

**Review by vjeffrey — COMMENTED** [review]
GitHub won't let me +1 this because i opened it,  but +1 , thank you for the updates! 🎉

File: .vscode/launch.json (hunk 1 above)
Comment (chris-rock) [reply .vscode/launch.json]: I 100% agree. Machine is destroyed, but anyway

**Review by arlimus — APPROVED** [review]
noice!

**Pull request merged**